velox/functions/sparksql/GetJsonObject.h

```diff
@@ -120,20 +118,77 @@ struct GetJsonObjectFunction {
     return result;
   }
 
-  // Normalizes the JSON path to be Spark-compatible:
-  // - Removes single quotes in bracket notation
-  // - Removes spaces after dots (e.g., "$. a" -> "$.a")
+  // Normalizes the JSON path to be Spark-compatible.
+  //
+  // Rules applied:
+  // 1. Removes single quotes in bracket notation (e.g., "$['a']" -> "$[a]").
+  // 2. Removes spaces after dots (e.g., "$. a" -> "$.a").
+  // 3. Removes trailing spaces after root symbol (e.g., "$ " -> "$").
+  // 4. Invalid cases return "-1":
```

Comment:
> Invalid cases return "-1":

This is a strange API choice. Why not change return type to `std::optional<std::string>` and return std::nullopt?